blockchain/fullblocktests/generate.go

```diff
@@ -1054,60 +1057,141 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 
 	// Attempt to add block with a regular tx in the stake tree.
 	//
-	//   ... -> b36(8)
+	//   ... -> bsl5(8)
 	//                \-> bv2(9)
-	g.SetTip("b36")
+	g.SetTip("bsl5")
 	g.NextBlock("bv2", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
 		b.STransactions[0] = b.Transactions[0]
 	})
 	rejected(blockchain.ErrRegTxInStakeTree)
 
 	// Attempt to add block with too many votes.
 	//
-	//   ... -> b36(8)
+	//   ... -> bsl5(8)
 	//                \-> bv3(9)
-	g.SetTip("b36")
+	g.SetTip("bsl5")
 	g.NextBlock("bv3", outs[9], ticketOuts[9], g.ReplaceWithNVotes(ticketsPerBlock+1))
 	rejected(blockchain.ErrTooManyVotes)
 
 	// Attempt to add block with too few votes.
 	//
-	//   ... -> b36(8)
+	//   ... -> bsl5(8)
 	//                \-> bv4(9)
-	g.SetTip("b36")
+	g.SetTip("bsl5")
 	g.NextBlock("bv4", outs[9], ticketOuts[9], g.ReplaceWithNVotes(ticketsPerBlock/2))
 	rejected(blockchain.ErrNotEnoughVotes)
 
 	// Attempt to add block with different number of votes in stake tree and
 	// header.
 	//
-	//   ... -> b36(8)
+	//   ... -> bsl5(8)
 	//                \-> bv5(9)
-	g.SetTip("b36")
+	g.SetTip("bsl5")
 	g.NextBlock("bv5", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
 		b.Header.FreshStake -= 1
 	})
 	rejected(blockchain.ErrFreshStakeMismatch)
 
+	// Attempt to add block with tickets voting on the wrong block.
+	//
+	//   ... -> bsl5(8)
+	//                \-> bv6(9)
+	g.SetTip("bsl5")
+	g.NextBlock("bv6", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		// invalidate the vote by corrupting the output script
+		b.STransactions[0].TxOut[0].PkScript[2] ^= 0x55
```

Comment:
This works, but I think a better test would be to actually replace the entire script with a completely valid script that votes on `bs14` (the block before the one it should be voting on which is `bs14`).

That way it will ensure it doesn't allow voting on the wrong block even when the block it's voting on is actually valid.